Torch/Utils/SteamWorkshopTools/SteamWorkshopTools.cs

```diff
@@ -0,0 +1,283 @@
+                allFilesDetails = await Task.Run(delegate {
+                    try
+                    {
+                        return remoteStorage.GetPublishedFileDetails1(
+                            itemcount: workshopIds.Count(), 
+                            publishedfileids: workshopIds, 
+                            method: HttpMethod.Post);
+                    }
+                    catch (HttpRequestException e)
+                    {
+                        Log.Error($"Fetching File Details failed: {e.Message}");
+                        return null;
+                    }
+                });
+                if (allFilesDetails == null)
+                    return null;
+                //fileDetails = remoteStorage.Call(HttpMethod.Post, "GetPublishedFileDetails", 1, new Dictionary<string, string>() { { "itemcount", workshopIds.Count().ToString() }, { "publishedfileids", workshopIds.ToString() } });
+                var detailsList = allFilesDetails?.Children.Find((KeyValue kv) => kv.Name == "publishedfiledetails")?.Children;
+                var resultCount = allFilesDetails?.GetValueOrDefault<int>("resultcount");
+                if( detailsList == null || resultCount == null)
+                {
+                    Log.Error("Received invalid data: ");
+#if DEBUG
+                    if(allFilesDetails != null)
+                        PrintKeyValue(allFilesDetails);
+                    return null;
+#endif
+                }
+                if ( detailsList.Count != workshopIds.Count() || resultCount != workshopIds.Count())
+                {
+                    Log.Error($"Received unexpected number of fileDetails. Expected: {workshopIds.Count()}, Received: {resultCount}");
+                    return null;
+                }
+
+                var result = new Dictionary<ulong, PublishedItemDetails>();
+                for( int i = 0; i < resultCount; i++ )
+                {
+                    var fileDetails = detailsList[i];
+
+                    var tagContainer = fileDetails.Children.Find(item => item.Name == "tags");
+                    List<string> tags = new List<string>();
+                    if (tagContainer != null)
+                        foreach (var tagKv in tagContainer.Children)
+                        {
+                            var tag = tagKv.Children.Find(item => item.Name == "tag")?.Value;
+                            if( tag != null)
+                                tags.Add(tag);
+                        }
+
+                    var publishedFileId = fileDetails.GetValueOrDefault<ulong>("publishedfileid");
+                    result[publishedFileId] = new PublishedItemDetails()
+                    {
+                        PublishedFileId = publishedFileId,
+                        Views           = fileDetails.GetValueOrDefault<uint>("views"),
+                        Subscriptions   = fileDetails.GetValueOrDefault<uint>("subscriptions"),
```

Comment:
This definitely looks like it can be replaced with JSON code models.

Reply:
Uh... Can you elaborate a little on this? Are you thinking about a specific library?

Reply:
The data itself is Steam's key-value format which is a pain to work with; I don't think there's a better way to do it here.

Reply:
I'll wait for another statement from @rexxar-tc before I act on it :)